packages/block-editor/src/store/reducer.js

```diff
@@ -381,6 +384,17 @@ const withBlockCache = ( reducer ) => ( state = {}, action ) => {
 		}
 	}
 
+	if ( newState.cache !== state.cache ) {
+		newState.getBlocksCache = {
```

Comment:
This is pretty badly unoptimized. I think we can make this a lot better by tweaking it on a per-action case, but I _think_ this lets us prevent the cache key from changing when using replaceInnerBlocks. (We will have to ensure that it does not change in more cases than just that, though. I'm not sure how feasible that will be. :/)